app/controllers/sentences_controller.php

```diff
@@ -1033,14 +1054,24 @@ public function several_random_sentences()
 
         $randomIds = $this->Sentence->getSeveralRandomIds($lang, $number);
 
-        $this->Session->write('random_lang_selected', $lang);
+        if(is_bool($randomIds) && !$randomIds) {
+            $searchDisabled = !Configure::read('Search.enabled');
+            $this->set('searchProblem', true);
+            if ($searchDisabled) {
```

Comment:
Well I will leave it to you and @jiru then :)

My comment was initially triggered by the fact that it felt strange to have `$searchDisabled` encapsulated inside `if (!$randomIds)` because if the search is disabled in the first place, you wouldn't even search for random ids.

The way things are now, you have a bit of redundancy since you're checking for Search.enable both in the model and in the controller.

But this is far from being a blocking issue to me. In the end the code does what's expected.